flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** Test for {@link CatalogLoader}. */
+public class TestCatalogLoader extends FlinkTestBase {
+
+  private static File warehouse = null;
+  private static final TableIdentifier IDENTIFIER = TableIdentifier.of("default", "my_table");
+  private static final Schema SCHEMA =
+      new Schema(Types.NestedField.required(1, "f1", Types.StringType.get()));
+
+  @BeforeClass
+  public static void createWarehouse() throws IOException {
+    warehouse = File.createTempFile("warehouse", null);
+    Assert.assertTrue(warehouse.delete());
+    hiveConf.set("my_key", "my_value");
+  }
+
+  @AfterClass
+  public static void dropWarehouse() throws IOException {
+    if (warehouse != null && warehouse.exists()) {
+      Path warehousePath = new Path(warehouse.getAbsolutePath());
+      FileSystem fs = warehousePath.getFileSystem(hiveConf);
+      Assert.assertTrue("Failed to delete " + warehousePath, fs.delete(warehousePath, true));
+    }
+  }
+
+  @Test
+  public void testHadoopCatalogLoader() throws IOException, ClassNotFoundException {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, "file:" + warehouse);
+    CatalogLoader loader = CatalogLoader.hadoop("my_catalog", hiveConf, properties);
+    validateCatalogLoader(loader);
+  }
+
+  @Test
+  public void testHiveCatalogLoader() throws IOException, ClassNotFoundException {
+    CatalogLoader loader = CatalogLoader.hive("my_catalog", hiveConf, Maps.newHashMap());
+    validateCatalogLoader(loader);
+  }
+
+  @Test
+  public void testRESTCatalogLoader() throws IOException, ClassNotFoundException {
+    CatalogLoader loader = CatalogLoader.rest("my_catalog", hiveConf, Maps.newHashMap());
+    validateCatalogLoader(loader);
+  }
+
+  private static void validateCatalogLoader(CatalogLoader loader)
+      throws IOException, ClassNotFoundException {
+    Table table = javaSerAndDeSer(loader).loadCatalog().createTable(IDENTIFIER, SCHEMA);
+    validateHadoopConf(table);
+  }
+
+  private static void validateHadoopConf(Table table) {
+    FileIO io = table.io();
+    Assertions.assertThat(io)
+        .as("FileIO should be a HadoopFileIO")
+        .isInstanceOf(HadoopFileIO.class);
+    HadoopFileIO hadoopIO = (HadoopFileIO) io;
+    Assert.assertEquals("my_value", hadoopIO.conf().get("my_key"));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <T> T javaSerAndDeSer(T object) throws IOException, ClassNotFoundException {
```

Comment:
nit: maybe just `javaSerdes` is enough as method name?

Reply:
Good one, thanks!